src/ansiblelint/utils.py

```diff
@@ -901,3 +898,22 @@ def nested_items(
             yield "list-item", item, parent
             for k, v, p in nested_items(item):
                 yield k, v, p
+
+
+def nested_items_path(
```

Comment:
I wonder if we can take this opportunity to move it to a dedicated `pathutils` module instead of keeping it in this wholesale utils dump.

Reply:
While I do not love the `utils` kitchensink, I do not think that the move is needed now, especially as we do not have an already existing module that would better encapsulate that type of code. That is not about file or paths, is about data parsing. Anyway let's avoid scope creeping, keeping the location is better, at least for now.

Reply:
> So, you're proposing `ansiblelint.path_utils` (or maybe `ansiblelint.data_utils` would be more descriptive?).

Something like that. It should be specific and accurate.
The problem with generic non-specific namespaces like `utils` or `helpers` is the same as with variables like `data`, `things` and `stuff` — they don't provide any clarity/insight as to why they exist, what's inside and what is their relation to the current level of abstraction. Over time, these such modules become dumps for helpers that do something needed somewhere but have no specific purpose.

Here's one of the explanations of this problem: https://tonsky.me/blog/utils/.

> Would you want to move any other functions to such a new module?

In general, yes. But not within this PR. It shouldn't contain things not related to the reimplementation of this one function.

> I think this function fits well with many of these other utils. It helps in finding/iterating through playbooks and tasks files. Is it worth it to split them into more modules?

See the comments above.

Reply:
As for "in the scope" vs not — I think it is because there is a deprecation warning that points at the new importable. It would be better for it to point at the final location so we wouldn't need to go through several rounds of deprecations as we decide where to better put this thing.

Alternatively, the deprecation bits could be put in a dedicated atomic PR where it could be coupled with the new proper selection.

Reply:
OK. I'm fine with moving it - `utils` is way too big. So what should we name it? Maybe `yaml_utils` would be most appropriate since this is primarily for iterating through yaml data and at least a few of the other utils relate to that as well (to be moved in a follow-up).

Reply:
Works for me. As long as it's accurate. Maybe mention the expectations in the module docstring so that non-matching things wouldn't appear there.

Reply:
K. I moved it to `yaml_utils`.

Reply:
> While I do not love the `utils` kitchensink, I do not think that the move is needed now, especially as we do not have an already existing module that would better encapsulate that type of code. That is not about file or paths, is about data parsing. Anyway let's avoid scope creeping, keeping the location is better, at least for now.

Oh. I missed this comment. What do you think of where it is now: `yaml_utils` ? Or should I move it back.